### ISSUE TITLE
ValueError reports that object does not exist when the object does actually exist, it is just not assigned to the content_type of the related object

### ISSUE THREAD

**Issue by jmcgill298** [nautobot/nautobot#4261]
### Environment
* Nautobot version (Docker tag too if applicable): 2.0.0b2
* Python version: 3.10
* Database platform, version: N/A
* Middleware(s): N/A

### Steps to Reproduce
1. Create a Role, but do not assign any ContentTypes to it
3. Create a Device with a role assigned to Role from step 1
4. Attempt to perform validated_save() on Device from step 2

### Expected Behavior
Error reporting that the Role assigned to the Device cannot be associated with dcim.device objects

### Observed Behavior
Error reporting that the Role does not exist (although it clearly does)

**Comment by HanlinMiao** [comment]
Should just be as easy as adding a try-except block in the `Device` Model's `clean()` method to catch this with a custom and helpful message without letting it fall to Django's default error handling.
```
def clean(self):
        from nautobot.ipam import models as ipam_models  # circular import workaround

        super().clean()
        # Validate Assigned Role
        from nautobot.extras.models import Role
        from django.core.exceptions import ObjectDoesNotExist

        if self.role is not None:
            try:
                Role.objects.get(id=self.role.id, content_types=ContentType.objects.get_for_model(Device))
            except ObjectDoesNotExist:
                raise ValidationError({"role": f"Assigned role {self.role} does not have Device ContentType assigned to it"})
```

**Comment by glennmatthews** [comment]
This is only one of many potential cases where an `ObjectDoesNotExist` could be raised due to content-type restrictions... IMHO it would be better to find a deeper level fix.

**Comment by jmcgill298** [comment]
I agree with @glennmatthews my example was only meant to provide a means of reproducing the issue, however the issue title was written to be generic to show that it is an issue throughout the codebase
